katsdpcontroller/test/test_product_config.py

```diff
@@ -1,73 +1,1012 @@
+        assert_equal(vis.flag_data_rate(1.0, 0), vis.flag_size / vis.int_time * 8)
+
+    def test_defaults(self) -> None:
+        del self.config['output_channels']
+        del self.config['excise']
+        vis = VisStream.from_config(Options(), 'sdp_l0', self.config, [self.bcp], {})
+        assert_equal(vis.output_channels, (0, 32768))
+        assert_equal(vis.excise, True)
+
+    def test_bad_continuum_factor(self) -> None:
+        self.config['continuum_factor'] = 3
+        with assert_raises_regex(
+                ValueError,
+                r'n_chans \(32768\) is not a multiple of required alignment \(12\)'):
+            VisStream.from_config(Options(), 'sdp_l0', self.config, [self.bcp], {})
+
+    def test_misaligned_output_channels(self):
+        self.config['continuum_factor'] = 2048
+        vis = VisStream.from_config(Options(), 'sdp_l0', self.config, [self.bcp], {})
+        assert_equal(vis.output_channels, (0, 8192))
+
+    def test_compatible(self) -> None:
+        vis1 = VisStream.from_config(Options(), 'sdp_l0', self.config, [self.bcp], {})
+        self.config['continuum_factor'] = 1
+        self.config['archive'] = True
+        vis2 = VisStream.from_config(Options(), 'sdp_l0', self.config, [self.bcp], {})
+        del self.config['output_channels']
+        vis3 = VisStream.from_config(Options(), 'sdp_l0', self.config, [self.bcp], {})
+        assert_true(vis1.compatible(vis1))
+        assert_true(vis1.compatible(vis2))
+        assert_false(vis1.compatible(vis3))
+
+    def test_develop(self) -> None:
+        options = Options(develop=True)
+        vis = VisStream.from_config(options, 'sdp_l0', self.config, [self.bcp], {})
+        assert_equal(vis.n_servers, 2)
+
+
+class TestBeamformerStream:
+    """Test :class:`~.BeamformerStream`."""
+
+    def setup(self) -> None:
+        self.acv = make_antenna_channelised_voltage()
+        self.tacv = [
+            make_tied_array_channelised_voltage(
+                self.acv, 'beam_0x', yarl.URL('spead://239.10.0.0+255:7148')),
+            make_tied_array_channelised_voltage(
+                self.acv, 'beam_0y', yarl.URL('spead://239.10.1.0+255:7148'))
+        ]
+        self.config = {
+            'type': 'sdp.beamformer',
+            'src_streams': ['beam_0x', 'beam_0y']
+        }
+
+    def test_from_config(self) -> None:
+        bf = BeamformerStream.from_config(Options(), 'beamformer', self.config, self.tacv, {})
+        assert_equal(bf.name, 'beamformer')
+        assert_equal(bf.antenna_channelised_voltage.name, 'narrow1_acv')
+        assert_equal(bf.tied_array_channelised_voltage, self.tacv)
+        assert_equal(bf.n_chans, 32768)
+
+    def test_mismatched_sources(self) -> None:
+        acv = make_antenna_channelised_voltage()
+        self.tacv[1] = make_tied_array_channelised_voltage(
+            acv, 'beam_0y', yarl.URL('spead://239.10.1.0+255:7148'))
+        with assert_raises_regex(ValueError,
```

Comment:
This test checks that the acv's are not the same object. Shouldn't it actually check that the actual config of the ACV is different ?

Reply:
Ok, I've made an ACV that is configured with different antennas.